src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryJsonConverter.cs

```diff
@@ -8,14 +8,39 @@ internal class FieldNameQueryJsonConverter<TReadAs> : ReserializeJsonConverter<T
 	{
 		protected override object DeserializeJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
 		{
-			//{
 			reader.Read(); //property name
-			var fieldName = reader.Value as string;
-			reader.Read(); //{
-			var query = ReadAs(reader, objectType, existingValue, serializer);
+			var fieldName = (string)reader.Value;
+			reader.Read();
+			TReadAs query = null;
 
-			if (query == null) return null;
+			switch (reader.TokenType)
+			{
+				case JsonToken.StartObject:
+					query = ReadAs(reader, objectType, existingValue, serializer);
+					break;
+				case JsonToken.Null:
+					break;
+				default:
+					query = new TReadAs();
+					switch (query)
+					{
```

Comment:
We need to cover all of these: 

https://github.com/elastic/clients-flight-recorder/blob/master/script/lib/normalizer.js#L430-L434

Reply:
```json
  const fieldQueries = [
    'range', 'common', 'intervals', 'match', 'match_bool_prefix', 'match_phrase', 'match_phrase_prefix',
    'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature',
    'term', 'terms'
  ]
```

I don't think the other cases are needed. It looks like 'range', 'common', 'intervals', 'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature' and 'terms' query do not support a query short form. The `switch` statement on type of query only applies when the value assigned to the field name key is not a JSON object.